src/lib/security/selinux_config.rb

```diff
@@ -25,92 +25,156 @@ module Security
 
+    # Known keys for setting a SELinux policy via kernel command line
     POLICY_KEYS = ["security", "selinux", "enforcing"].freeze
     private_constant :POLICY_KEYS
 
+    # Known SELinux policies
+    #
+    # This is _the main_ or _base_ configuration for known SELinux policies. However, note that, for
+    # example, a permissive mode could be set by just setting the "security" module; i.e.,
+    # "security=selinux" means "enable SELinux using the permissive mode". Or even setting
+    # the enforcing param to a value equal or less than 0; i.e., "security=selinux enforcing=0".
+    #
+    # Additionally, removing the "security" from the kernel params does not mean to use none
+    # security module. Instead, it just fallback to the kernel configuration at the compile time,
+    # which in SUSE is to use AppArmor according to the CONFIG_LSM variable. So, it could be said
+    # that dropping all policy param to disabling SELinux is safe enough.
+    #
+    # To know more, please visit the LSM Usage documentation at
+    # https://www.kernel.org/doc/html/latest/admin-guide/LSM/index.html
+    # and/or grep for CONFIG_LSM in /boot/config-*
     POLICIES = {
+      # NOTE: setting the "security" module to :missing, which means to remove the options from the
+      #       kernel
       disabled:   { "security" => :missing, "selinux" => :missing, "enforcing" => :missing },
       permissive: { "security" => "selinux", "selinux" => "1", "enforcing" => :missing },
       enforcing:  { "security" => "selinux", "selinux" => "1", "enforcing" => "1" },
     }.freeze
     private_constant :POLICIES
 
-    def configured_policy
-      match_policy(policy_options_from_kernel) || :disabled
+    # Returns the requested policy, if exists
+    #
+    # @param key [String, Symbol] the policy identifier
+    # @return [Hash, nil] options for matched policy or nil if none
+    def find_policy(key)
+      POLICIES[key.to_sym]
     end
 
-    # Propose a policy according to the value set in the control-file
+    # Proposes a policy based on `selinux_policy` value set in the control file
     def propose_policy
-      key = :enforcing # read it from control-file
+      policy_key = Yast::ProductFeatures.GetFeature("globals", "selinux_policy").to_sym
 
-      log.info "Proposing the `#{key}` SELinux policy"
-      update_policy(policy_key)
+      log.info "Proposing the `#{policy_key}` SELinux policy"
+      self.policy = policy_key
       save
     end
 
-    def find_policy(key)
-      POLICIES[key]
-    end
+    # Returns the configured SELinux policy according to params in kernel command line
+    #
+    # @see #policy_from_kernel_params
+    #
+    # @return [Symbol] the policy identifier
+    def configured_policy
+      options = policy_from_kernel_params
+
+      return :disabled if options.empty?
+
+      security_module = options["security"]
+      module_disabled = options["selinux"].to_i <= 0
+      enforcing_mode  = options["enforcing"].to_i > 0
```

Comment:
so e.g. enforcing=15 is valid?

Reply:
Yes, it is.

![Screenshot_opensusetumbleweed_2021-02-02_10:36:06](https://user-images.githubusercontent.com/1691872/106588276-81829e00-6542-11eb-9ddf-f7aee35507b1.png)